### PR TITLE
feat: remove budget and toast #P23-42

### PR DESCRIPTION
**Pull request by Alessiass**
## Overview 📖 

Link to task: https://tracker.intive.com/jira/browse/PATRO23-42
User can now delete a budget. If removal is succefull Toast component pops up and user is redirected to main page. If error occurs the delete window is closed with information that something went wrong.

## Toast 🍞 
How to use Toast (thanks to @wlechowicz for support with managing global state of it).

First we import the useToast hook:
```
import { useToast } from "ui";

const showToast = useToast();
```

Then we just call it by passing proper message and variant (confirm or error):
```
showToast({
        variant: "error",
        message: "There is no spoon",
      });
```

And that's it! Since Toast component is added to navigation layout it cannot be called outside of it.

### CODE REVIEW

**Comment by wlechowicz** [comment]
> Since Toast component is added to navigation layout it cannot be called outside of it.

Clarification, ToastHoast component can be added to (regflow) layout, but I didn't do it, since we don't have such a use case.

File: apps/web/lib/dictionary.tsx [hunk 1]
```diff
@@ -711,6 +711,33 @@ const dictionary = {
       },
       dateFormats: { en: "en-GB", pl: "pl", fr: "fr-FR" },
     },
+    removeBudgetModal: {
+      header: {
+        en: "Are you sure you want to delete the budget:",
+        pl: "Czy na pewno chcesz usunąć ten budżet:",
+        fr: "Souhaitez-vous hon hon supprimer ce uoi budget:",
```
Comment (wlechowicz) [inline apps/web/lib/dictionary.tsx:718]: @zuzannapaw 🐸

**Review by wlechowicz — APPROVED** [review]

**Review by monika-klokosinska-intive — APPROVED** [review]

File: apps/web/app/(navigation)/budgets/[id]/BudgetBasicInformation.tsx [hunk 2]
```diff
@@ -83,6 +77,9 @@ export function BudgetBasicInformation({ budget }: BudgetBasicInfoProps) {
     setIsEditBudgetModalOpen(false);
   };
 
+  const [deleteModalVisibility, setDeleteModalVisibility] =
+    useState<boolean>(false);
```
Comment (aniasgit) [inline apps/web/app/(navigation)/budgets/[id]/BudgetBasicInformation.tsx:81]: If you set useState false, you don't have to type it to boolean, the type is set automatically.

File: apps/web/app/(navigation)/budgets/[id]/RemoveBudget.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+export const RemoveBudget = ({ budget, onClose }: RemoveBudgetProps) => {
+  const { t, dict } = useTranslate("BudgetsPage");
+  const queryClient = useQueryClient();
+  const { data: session } = useSession();
+  const { replace } = useRouter();
+  const showToast = useToast();
+
+  const deleteBudget = useMutation({
+    mutationFn: (id: string) => {
+      return fetch(env.NEXT_PUBLIC_API_URL + "/budgets/" + id, {
```
Comment (aniasgit) [inline apps/web/app/(navigation)/budgets/[id]/RemoveBudget.tsx:30]:
```suggestion
      return fetch(`${env.NEXT_PUBLIC_API_URL}/budgets/${ id}`, {
```

File: apps/web/package.json [hunk 4]
```diff
@@ -14,7 +14,8 @@
     "@tanstack/react-query": "^4.29.5",
     "chart.js": "^4.3.0",
     "dayjs": "^1.11.7",
-    "houseform": "^1.9.0",
+    "houseform": "^1.10.5",
```
Comment (aniasgit) [inline apps/web/package.json:17]: In houseform 1.10 clearing the datePicker crashes the application, so we keep going back to version 1.9, don't bump it!

File: packages/ui/Toast/Toast.styles.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  @media (min-width: 768px) {
+    padding: 24px;
+  }
+`;
+
+export const StylingDiv = styled.div<ToastVariant>`
+  background-color: ${({ theme, variant }) => theme.toast[variant].background};
+  color: ${({ theme, variant }) => theme.toast[variant].main};
+  border: 2px solid ${({ theme, variant }) => theme.toast[variant].main};
+`;
```
Comment (aniasgit) [inline packages/ui/Toast/Toast.styles.ts:36]: I'd love to see variants of toast in the storybook. I know it's a matter of taste, but I don't like the current colors of the "confirm" variant, the pink background looks weird. I'd rather go for something like this but with border: 
alert

**Review by bdeluga — APPROVED** [review]
LGTM 🚀

**Pull request merged**